### PR TITLE
[docs] - Getting Started/quick start updates [CON-39]

### PR DESCRIPTION
**Pull request by erinkcochran87**
### Summary & Motivation

This PR updates the Getting Started (quick start):

- Adds Python requirement to installation step
- Adds mention of assets
- Reformats the **Running the job** section
- Adds a **What's next?** section with suggestions and links for next steps

### CODE REVIEW

File: docs/content/getting-started.mdx [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
 
-This will install the latest stable version of the core Dagster packages in your current Python environment.
+This installs the latest stable version of the core Dagster packages in your current Python environment.
+
+---
+
+## Writing a job
 
-## Writing a Job
+To get started, we'll write some simple [ops](/concepts/ops-jobs-graphs/ops) and use them to create a [job](/concepts/ops-jobs-graphs/jobs-graphs). Dagster also offers a way to produce data artifacts like tables and files using [software-defined assets](/concepts/assets/software-defined-assets), but for simplicity we'll focus on ops.
```
Comment (sryza) [inline docs/content/getting-started.mdx:27]: I think we probably want to focus on assets here instead of ops.

**Comment by erinkcochran87** [comment]
https://github.com/dagster-io/dagster/pull/7804 Could be useful here.

**Comment by erinkcochran87** [comment]
Trying something new here - scrapping the quick start and making the front page about Dagster and its value. This builds on the work from https://github.com/dagster-io/dagster/pull/7804, which I think was a big improvement over the current GS.

My thinking is that this better communicates what Dagster is, why you'd use it, what you can do with it, and where to go next. [Preview is here](https://dagster-git-getting-started-updates-elementl.vercel.app/master/getting-started).

If this approach is well-received, I'll work on updating the copy and re-submitting for review.

**Comment by sryza** [comment]
I like this direction a lot. @schrockn had recommended putting the breaks on my last PR, which took a similar approach, because we're about to embark on a positioning exercise, and he thought it might make sense to base this content on what came out of that.  That said, I think that the sooner we can make a change like this, the better.  It might be worth seeing if we can spend an hour this week to try to get to alignment on what should show up here. What I had put down in my last version is a little bit radical, because it explains Dagster purely in terms of assets, so I think we need to decide whether / how much to tone that down.

Thoughts @schrockn?

**Comment by erinkcochran87** [comment]
I agree completely @sryza - we definitely want to get the positioning right.

If this exercise doesn't fit into the timing for 0.15.0, I think we could release a version of this that isn't quite so asset-heavy but still focuses on product introduction/orientation. Thoughts on that would also be helpful.

File: docs/content/getting-started.mdx [hunk 2]
```diff
@@ -1,98 +1,126 @@
...
 ```
 
-Save the code above in a file named `hello_world.py`.
+---
 
-You can execute the job in three different ways: [Dagit](/concepts/dagit/dagit), [Dagster Python API](/concepts/ops-jobs-graphs/job-execution#python-apis), or [Dagster CLI](/\_apidocs/cli#dagster-pipeline-execute).
+## Step 3: Materialize the asset
 
-## Running the Job in Dagit
+Next, you'll materialize the asset. Materialization computes an asset's contents and writes them to persistent storage.
```
Comment (sryza) [inline docs/content/getting-started.mdx:58]: I think it's worth being specific about what will happen here.  E.g. "Materializing computes an asset's contents and writes them to persistent storage - by default, a pickle file on the local filesystem."

File: docs/content/getting-started.mdx [hunk 3]
```diff
@@ -1,98 +1,126 @@
...
+```python file=/guides/dagster/asset_tutorial/cereal.py startafter=start_asset_marker endbefore=end_asset_marker
+import csv
+import requests
+from dagster import asset
 
-@op
-def hello(name: str):
-    print(f"Hello, {name}!")
 
+@asset
```
Comment (sryza) [inline docs/content/getting-started.mdx:45]: The prior example had a dependency between two ops, and I think demonstrating that Dagster models dependencies is worth preserving.

We could use the example from the second page of the tutorial for this? https://docs.dagster.io/guides/dagster/asset-tutorial/asset-graph#lets-get-serial

**Review by sryza — COMMENTED** [review]
The structure and framing here look great. I left a couple granular comments on the content.

File: docs/content/getting-started.mdx (hunk 3 above)
Comment (yuhan) [reply docs/content/getting-started.mdx:45]: agreed. having a single asset here could confuse new users of why AssetGroup is needed at all. adding an additional downstream asset sounds great to me

File: docs/content/getting-started.mdx [hunk 4]
```diff
@@ -84,15 +93,15 @@ There are a few ways to materialize an asset:
...
 
-4. Click the **Materialize All** button to launch a run that materializes the asset:
+4. Click the **Materialize All** button to launch a run that materializes the assets:
 
    <img src="/images/getting-started/materialize-asset-in-dagit.gif" />
 
 ### Using the Dagster Python API
 
-You can also use the [Dagster Python API](/concepts/ops-jobs-graphs/job-execution#python-apis) materialize the asset as a script.
+You can also use the [Dagster Python API](/concepts/ops-jobs-graphs/job-execution#python-apis) materialize the assets as a script.
```
Comment (sryza) [inline docs/content/getting-started.mdx:104]: It looks like the Python example below only references `cereals`, not `nabisco_cereals`.

**Review by sryza — COMMENTED** [review]
Had one last comment. Also, should we get rid of getting_started/hello_world.py because it's not used anywhere? Otherwise, looks great.

**Comment by schrockn** [comment]
Hey @sryza in terms of the get started I think it is fine to lead with assets in this fashion. IMO there should be a link to the bottom that points to a page that describes how Dagster can be used for task-based orchestration as well. The question is how to word that so that it feels like a layering rather than two parallel systems.

**Comment by schrockn** [comment]
Hey @sryza in terms of the get started I think it is fine to lead with assets in this fashion. IMO there should be a link to the bottom that points to a page that describes how Dagster can be used for task-based orchestration as well. The question is how to word that so that it feels like a layering rather than two parallel systems.

**Review by sryza — APPROVED** [review]
I don't love this for the reasons we've discussed before, that apply to the prior version as well. Just to put them down here for the record, it assumes that the first thing someone wants to do when they get to our docs site is get Dagster running on their own computer. But I suspect that the first thing most people want to do is understand what Dagster _is_.  I don't think this change makes things any worse though on that dimension, and it puts assets front & center, so 👍 👍 👍 .

**Comment by erinkcochran87** [comment]
@sryza I agree completely, and we'll address that in the very near future.

**Pull request merged**